Source/JavaScriptCore/runtime/JSObject.cpp

```diff
@@ -4078,4 +4079,74 @@ TransitionKind JSObject::suggestedArrayStorageTransition() const
     return TransitionKind::AllocateArrayStorage;
 }
 
+void JSObject::putOwnDataPropertyBatching(VM& vm, const RefPtr<UniquedStringImpl>* properties, const EncodedJSValue* values, unsigned size)
+{
+    unsigned i = 0;
+    Structure* structure = this->structure();
+    if (!(structure->isDictionary() || (structure->transitionCountEstimate() + size) > Structure::s_maxTransitionLength || !structure->canPerformFastPropertyEnumeration() || (structure->transitionWatchpointSet().isBeingWatched() && structure->transitionWatchpointSet().isStillValid()))) {
```

Comment:
Are we only comparing `structure->transitionCountEstimate() + size)` against `Structure::s_maxTransitionLength` because it is smaller (and therefore, more conservative than) `s_maxTransitionLengthForNonEvalPutById`?  If so, I recommend that you add a `static_assert(Structure::s_maxTransitionLength < Structure::s_maxTransitionLengthForNonEvalPutById, "We should use the more conservative max transition length")` above here to document this.

Reply:
Sounds good, changed.

Reply:
Ah, we do not need s_maxTransitionLengthForNonEvalPutById. We always use s_maxTransitionLength since this is for C++ usage (not from JS function). So we can just use `s_maxTransitionLength`